Runtime/Scripts/Types/RGClientConnection_Remote.cs

```diff
@@ -114,113 +143,116 @@ public override bool Connected()
 
         public override async void Connect()
         {
+            // trickyness to get around can't call unity webrequest off of main thread
+            // or await inside lock statements
+            var shouldIConnect = false;
             if (!_connecting && !Connected())
+            {
+                lock (this)
+                {
+                    if (!_connecting && !Connected())
+                    {
+                        _connecting = true;
+                        shouldIConnect = true;
+                    }
+                }
+            }
```

Comment:
I've never used Lazy for a primitive value before. In c# primitives are atomic for a single operation.  There are still limits to cross thread visibility, especially in multi processor systems, but as long as you keep those conditions in mind you can avoid the overhead of Lazy's lock for everything pattern.

I avoided using lazy here as `ExecutionAndPublication` always locks when checking its value and the intent of the double checked locking pattern is to avoid locking when not necessary.  As I understand it, it would be roughly equivalent to Lazy's `ExecutionAndPublication` from a locking perspective to just lock and then check instead of using double checked locking; not exactly the same cpu cost, but logically similar.

In this particular case you could argue that double checked locking is only truly safe for preventing 2 simultaneous connection attempts, but may possibly miss `_connecting` transitioning to false as set by another thread and not connect in this method call when it technically could.  This statement is true... however, the system will still see on the next pass (RGBotServerListener.Update) that we aren't connected and will attempt the connection again.  This outer code already exists to detect random disconnect scenarios and is not specific to interactions with this locking code.

Long story short, I updated the code to use a Semaphore here instead of a boolean.  It doesn't change anything from a functionality perspective as described in the prior sentences, but it may be a bit more clear for others reading the code.